addons/sale_project/models/sale_order_line.py

```diff
@@ -277,13 +277,45 @@ def _timesheet_create_task_prepare_values(self, project):
             'user_ids': False,  # force non assigned task, as created as sudo()
         }
 
+    @api.model
+    def _get_product_service_policy(self):
+        return ['ordered_prepaid']
+
+    def _prepare_task_template_vals(self, template, project):
+        if template.allocated_hours:
+            allocated_hours = template.allocated_hours
+        else:
+            allocated_hours = sum(
+                sol._convert_qty_company_hours(self.company_id)
+                for sol in self.order_id.order_line
+                if sol.product_id.task_template_id.id == template.id
+                and sol.product_id.service_policy in self._get_product_service_policy()
+            )
+
+        return {
+            'name': '%s - %s' % (self.order_id.name, template.name),
+            'allocated_hours': allocated_hours,
+            'project_id': project.id,
+            'sale_line_id': self.id,
+            'sale_order_id': self.order_id.id,
+        }
+
+    def _get_sale_order_partner_id(self, project):
+        return self.order_id.partner_id.id
+
     def _timesheet_create_task(self, project):
         """ Generate task for the given so line, and link it.
             :param project: record of project.project in which the task should be created
             :return task: record of the created task
         """
-        values = self._timesheet_create_task_prepare_values(project)
-        task = self.env['project.task'].sudo().create(values)
+        if template := self.product_id.task_template_id:
+            vals = self._prepare_task_template_vals(template, project)
+            task_id = template.action_create_from_template(vals)
+            task = self.env['project.task'].sudo().browse(task_id)
+            task.sudo().partner_id = self._get_sale_order_partner_id(project)
```

Comment:
Hello @xavierbol 

When we copy a task template, the partner is removed because partner_id is added to the blacklist. Therefore, I need to write the partner ID separately [link](https://github.com/odoo/odoo/blob/master/addons/project/models/project_task.py#L1982). If the task is FSM, it returns the partner_shipping_id.